src/Reference.php

```diff
@@ -74,21 +75,21 @@ public function __construct($link)
     }
 
     /**
-     * Will use #ref_<link>.
-     *
-     * @return string
+     * Initialization.
      */
-    public function getDesiredName()
+    public function init()
     {
-        return '#ref_'.$this->link;
+        $this->_init();
     }
 
     /**
-     * Initialization.
+     * Will use #ref_<link>.
+     *
+     * @return string
      */
-    public function init()
+    public function getDesiredName()
     {
-        $this->_init();
+        return '#ref_'.$this->link;
```

Comment:
shuffle :D

Reply:
yeah, sorry. I just like that __constructor and init() methods are always defined first in class code :)